src/service/resources-impl.js

```diff
@@ -600,6 +601,22 @@ export class Resources {
     });
   }
 
+  collapseElement(element) {
+    const box = this.viewport_.getLayoutRect(element);
+    const resource = Resource.forElement(element);
+    if (box.width != 0 && box.height != 0) {
+      this.setRelayoutTop_(box.top);
```

Comment:
A performance question: if `collapseElement` is frequently called as the result of change height - would this be an extra work to relayout from the box.top again? I actually think in reality it won't be the case because this callback will be executed before the next pass. If that's true, then we should be all good.